packages/dd-trace/src/plugins/plugin.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const dc = require('diagnostics_channel')
+const { storage } = require('../../../datadog-core')
+
+class Subscription {
+  constructor (event, handler) {
+    this._channel = dc.channel(event)
+    this._handler = handler
+  }
+
+  enable () {
+    this._channel.subscribe(this._handler)
+  }
+
+  disable () {
+    this._channel.unsubscribe(this._handler)
+  }
+}
+
+module.exports = class Plugin {
+  constructor () {
+    this._subscriptions = []
+    this._enabled = false
+    this._storeStack = []
+  }
+
+  enter (span, store) {
+    store = store || storage.getStore()
```

Comment:
This now this is always called with `storage.getStore()`. Is it even needed to pass a store explicitly? Shouldn't you always be in the proper store before entering a span?

Reply:
It's needed. We need to create a span with `startSpan` before going through with `enter`. We need the stored span for the `childOf`.

Reply:
The value that is passed is _always_ `storage.getStore()` anyway, so why accept this parameter at all? Why not just always use `storage.getStore()`?